diff --git a/jitterentropy.service.in b/jitterentropy.service.in
index ea47b7a..a786448 100644
--- a/jitterentropy.service.in
+++ b/jitterentropy.service.in
@@ -19,7 +19,7 @@ DeviceAllow=/dev/random rw
 DevicePolicy=strict
 ExecStart=@PATH@/jitterentropy-rngd
 IPAddressDeny=any
-LimitMEMLOCK=0
+LimitMEMLOCK=2M
 LockPersonality=yes
 MemoryDenyWriteExecute=yes
 MountFlags=private
@@ -41,7 +41,7 @@ RestrictNamespaces=yes
 RestrictRealtime=yes
 SystemCallArchitectures=native
 SystemCallFilter=@system-service
-SystemCallFilter=~@chown @clock @cpu-emulation @debug @ipc @module @mount @obsolete @privileged @raw-io @reboot @resources @swap memfd_create mincore mlock mlockall personality
+SystemCallFilter=~@chown @clock @cpu-emulation @debug @ipc @module @mount @obsolete @privileged @raw-io @reboot @resources @swap memfd_create mincore personality
 UMask=0077
 
 [Install]
