diff --git a/common/sampling.h b/common/sampling.h
index eeaa53b8..30d8f3df 100644
--- a/common/sampling.h
+++ b/common/sampling.h
@@ -24,12 +24,12 @@ typedef struct llama_sampling_params {
     int32_t     n_prev                = 64;                 // number of previous tokens to remember
     int32_t     n_probs               = 0;                  // if greater than 0, output the probabilities of top n_probs tokens.
     int32_t     min_keep              = 0;                  // 0 = disabled, otherwise samplers should return at least min_keep tokens
-    int32_t     top_k                 = 40;                 // <= 0 to use vocab size
-    float       top_p                 = 0.95f;              // 1.0 = disabled
-    float       min_p                 = 0.05f;              // 0.0 = disabled
+    int32_t     top_k                 = 0;                  // <= 0 to use vocab size
+    float       top_p                 = 1.00f;              // 1.0 = disabled
+    float       min_p                 = 0.00f;              // 0.0 = disabled
     float       tfs_z                 = 1.00f;              // 1.0 = disabled
     float       typical_p             = 1.00f;              // 1.0 = disabled
-    float       temp                  = 0.80f;              // <= 0.0 to sample greedily, 0.0 to not output probabilities
+    float       temp                  = 1.00f;              // <= 0.0 to sample greedily, 0.0 to not output probabilities
     float       dynatemp_range        = 0.00f;              // 0.0 = disabled
     float       dynatemp_exponent     = 1.00f;              // controls how entropy maps to temperature in dynamic temperature sampler
     int32_t     penalty_last_n        = 64;                 // last n tokens to penalize (0 = disable penalty, -1 = context size)
