From 3960ca554aee338a7f65f3abd17f0f56b433a918 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Thu, 4 May 2017 14:45:59 -0400
Subject: [PATCH 064/103] enable SECURITY_PERF_EVENTS_RESTRICT by default

Signed-off-by: Daniel Micay <danielmicay@gmail.com>
---
 security/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/security/Kconfig b/security/Kconfig
index ea669f5e261..731c88508ab 100644
--- a/security/Kconfig
+++ b/security/Kconfig
@@ -75,6 +75,7 @@ config MSEAL_SYSTEM_MAPPINGS
 config SECURITY_PERF_EVENTS_RESTRICT
 	bool "Restrict unprivileged use of performance events"
 	depends on PERF_EVENTS
+	default y
 	help
 	  If you say Y here, the kernel.perf_event_paranoid sysctl
 	  will be set to 3 by default, and no unprivileged use of the
-- 
2.52.0

