diff --git a/src/launch/launcher.c b/src/launch/launcher.c
index 5985712..a7d6447 100644
--- a/src/launch/launcher.c
+++ b/src/launch/launcher.c
@@ -719,13 +719,27 @@ static int launcher_load_service_file(Launcher *launcher, const char *path, cons
                         if (r)
                                 return error_trace(r);
                 } else {
-                        log_append_here(&launcher->log, LOG_ERR, 0, DBUS_BROKER_CATALOG_SERVICE_INVALID);
-                        log_append_service_path(&launcher->log, path);
-                        log_append_service_name(&launcher->log, name);
+                        char *old_real_path, *new_real_path;
+                        int difference;
+                        old_real_path = realpath(old_service->data->path, NULL);
+                        new_real_path = realpath(path, NULL);
+                        difference = strcmp(
+                                old_real_path ? old_real_path : old_service->data->path,
+                                new_real_path ? new_real_path : path);
+                        free(new_real_path);
+                        free(old_real_path);
+
+                        if (difference != 0) {
+                                log_append_here(&launcher->log, LOG_ERR, 0, DBUS_BROKER_CATALOG_SERVICE_INVALID);
+                                log_append_service_path(&launcher->log, path);
+                                log_append_service_name(&launcher->log, name);
 
-                        r = log_commitf(&launcher->log, "Ignoring duplicate name '%s' in service file '%s'\n", name, path);
-                        if (r)
-                                return error_fold(r);
+                                r = log_commitf(&launcher->log,
+                                                "Ignoring duplicate name '%s' in service file '%s' (already seen in '%s')\n",
+                                                name, path, old_service->data->path);
+                                if (r)
+                                        return error_fold(r);
+                        }
 
                         return LAUNCHER_E_INVALID_SERVICE_FILE;
                 }
